src/llm_utils.py

```diff
@@ -5,178 +5,136 @@
+        prompt = f"""
+        Analyze these code changes and provide feedback using EXACTLY this format:
 
-    Make sure to insert '---' in between two review comments. IT IS A MUST!!!
+        FILE: [file-path]
+        LINES: [start_line-end_line]  # MUST be within the same diff hunk
+        COMMENT: [clear explanation]
+        SUGGESTION: |
+        [exact code replacement OR N/A]
 
-    Focus on:
-    - Code quality issues
-    - Security vulnerabilities
-    - Performance optimizations
-    - Style inconsistencies
-    - Possible bugs
-    - Missing documentation
+        ---
+        Example Valid Comment:
+        FILE: src/utils.py
+        LINES: 15-18
+        COMMENT: Missing error handling for database connection
+        SUGGESTION: |
+        try:
+            db.connect()
+        except ConnectionError as e:
+            logger.error(f"Connection failed: {{e}}")
 
-    Diff content:
-    {diff_content}
-    """
+        ---
+        Invalid Example (will be rejected):
+        FILE: src/utils.py
+        LINES: fifteen-eighteen
+        COMMENT: This will be skipped
 
-    client = genai.Client(api_key=API_KEY)
+        ---
 
-    #Get analysis from Gemini model
-    response = client.models.generate_content(
-        model="gemini-2.0-flash", contents=prompt
-    )   
+        Important Rules:
+        1. Never include code snippets in COMMENT
+        2. Always verify line numbers exist in the diff
+        3. Keep suggestions focused on changed lines
+        4. Use exactly one '---' between comments
 
-    #print(response.text)
-    
-    # Parse and format the response
-    review_comments = parse_llm_response(response.text)
-    return review_comments
+        Diff content:
+        {diff_content}
+        """
 
+        client = genai.Client(api_key=API_KEY)
+
+        #Get analysis from Gemini model
+        response = client.models.generate_content(
+            model="gemini-2.0-flash", contents=prompt
+        )   
+
+        #print(response.text)
+        
+        # Parse and format the response
+        review_comments = parse_llm_response(response.text)
+        print(f"Parsed {len(review_comments)} valid comments")
+        return review_comments
+    except Exception as e:
+        print(f"LLM analysis failed: {str(e)}")
+        return []
 
 
-"""
-This is the final code given by deepseek to parse the response from the LLM model. It includes errors
-"""
 
 def parse_llm_response(response: str) -> List[Dict]:
+    """Parse LLM response with strict validation"""
     review_comments = []
     
-    # Split response into valid comment blocks
-    comment_blocks = response.split('---')
-    
-    for block in comment_blocks:
-        block = block.strip()
-        if not block:
+    for block in response.split('---'):
+        lines = [line.strip() for line in block.split('\n') if line.strip()]
+        if not lines:
             continue
-            
-        comment_data = {
+
+        comment = {
```

Comment:
Initialize comment
```suggestion
|
```
N/A